discovery/test_server.go

```diff
@@ -0,0 +1,55 @@
+package discovery
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+)
+
+// TestServer represents a Consul server we can run our tests against. Depends
+// on a local `consul` binary installed into our environ's PATH.
+type TestServer struct {
+	cmd      *exec.Cmd
+	HTTPAddr string
+}
+
+// NewTestServer constructs a new TestServer by including the httpPort as well.
+func NewTestServer(httpPort int) (*TestServer, error) {
+	path, err := exec.LookPath("consul")
+	if err != nil || path == "" {
+		return nil, fmt.Errorf("consul not found on $PATH - download and install " +
+			"consul or skip this test")
+	}
+
+	args := []string{"agent", "-dev"}
+	cmd := exec.Command("consul", args...)
+	cmd.Stdout = io.Writer(os.Stdout)
+	cmd.Stderr = io.Writer(os.Stderr)
+	if err := cmd.Start(); err != nil {
+		return nil, errors.New("failed starting command")
+	}
+
+	httpAddr := fmt.Sprintf("127.0.0.1:%d", httpPort)
```

Comment:
You can inline this in the struct definition.

```go
return &TestServer{cmd, fmt.Sprintf("127.0.0.1:%d", httpPort)}, nil
```

Reply:
I have a personal preference around inlining. Normally I enjoy the extra bits for readability and searching. In this case I agree, the line can be kept fairly short. 👍